### PR TITLE
Add daily min temperatures as demo dataset for time series

### LINKED ISSUE #1775: Add daily min temperature demo dataset
- For time series problems, we should have a demo dataset that we can use in our documentation to show an example
- We should use the daily min temperatures as one of the datasets
  - https://github.com/alteryx/evalml/pull/3023

### PR DESCRIPTION
**Pull request by gsheni**
- Fixes #1775

### CODE REVIEW

**Comment by rwedge** [comment]
`load_weather` should be added to the Demo Datasets section of the API reference

**Comment by gsheni** [comment]
@rwedge added: https://github.com/alteryx/featuretools/pull/1777/commits/bd6c69082768236069501bc37e53616b90a18705

File: featuretools/demo/weather.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import pandas as pd
+
+import featuretools as ft
+
+
+def load_weather(nrows=None,
+                 return_single_table=False):
+    '''
+    Load the Australian daily-min-termperatures weather dataset.
```
Comment (rwedge) [inline featuretools/demo/weather.py:9]: temperatures typo

**Review by rwedge — APPROVED** [review]
LGTM

**Review by tamargrey — APPROVED** [review]
lgtm!

**Pull request merged**